src/services/reader/web_sys.rs

```diff
@@ -0,0 +1,160 @@
+//! `web-sys` implementation for the reader service.
+
+use crate::callback::Callback;
+use crate::services::Task;
+use gloo::events::EventListener;
+use js_sys::Uint8Array;
+use std::cmp;
+use std::fmt;
+pub use web_sys::{Blob, File};
+use web_sys::{Event, FileReader};
+
+/// Struct that represents data of a file.
+#[derive(Clone, Debug)]
+pub struct FileData {
+    /// Name of loaded file.
+    pub name: String,
+    /// Content of loaded file.
+    pub content: Vec<u8>,
+}
+
+/// Struct that represents a chunk of a file.
+#[derive(Clone, Debug)]
+pub enum FileChunk {
+    /// Reading of chunks started. Equals **0%** progress.
+    Started {
+        /// Name of loaded file.
+        name: String,
+    },
+    /// The next data chunk that read. Also provides a progress value.
+    DataChunk {
+        /// The chunk of binary data.
+        data: Vec<u8>,
+        /// The progress value in interval: `0 < progress <= 1`.
+        progress: f32,
+    },
+    /// Reading of chunks finished. Equals **100%** progress.
+    Finished,
+}
```

Comment:
Please move to `reader.rs`, we can share these parts between `std_web` and `web_sys` implementations

Reply:
Done. I moved a couple of other things that I thought to be duplicates, let me know what you think.

Reply:
Yeah, looks great!